uk.co.compucorp.civicrm.tasksassignments/tests/phpunit/CRM/Tasksassignments/BAO/KeyDates.php

```diff
@@ -20,24 +20,19 @@ public function setUpHeadless() {
   }
   
   public function testDatesForDeletedContactsNotInResult() {
-    $contact = ContactFabricator::fabricate(['birth_date' => '1977-06-27']);
-    ContactFabricator::fabricate(['birth_date' => '1978-06-19']);
-    ContactFabricator::fabricate(['birth_date' => '1982-08-16']);
-
-    $query = '
-      UPDATE civicrm_contact
-      SET is_deleted = 1
-      WHERE id = %1
-    ';
-    $params = [1 => [$contact['id'], 'Int']];
-    CRM_Core_DAO::executeQuery($query, $params);
+    ContactFabricator::fabricate(['birth_date' => '1977-06-27', 'is_deleted' => 1]);
+    $contacts[] = ContactFabricator::fabricate(['birth_date' => '1978-06-19'])['id'];
+    $contacts[] = ContactFabricator::fabricate(['birth_date' => '1982-08-16'])['id'];
 
     $keyDates = civicrm_api3('KeyDates', 'get', [
       'sequential' => 1,
     ]);
 
+    $this->assertEquals(sizeof($contacts), $keyDates['count']);
+
+    // Verify deleted contact is not in result
     foreach ($keyDates['values'] as $currentDate) {
-      $this->assertNotEquals($contact['id'], $currentDate['contact_id']);
+      $this->assertContains($currentDate['contact_id'], $contacts);

```

Comment:
We can actually force the API to return in the order we want using the sort parameter.
Not just comfortable using assertContains on an array of arrays, its more prone to mistakes. It's much easier when its a single value against a simple array.